cluster-autoscaler/utils/drain/drain.go

```diff
@@ -287,3 +296,22 @@ func hasSafeToEvictAnnotation(pod *apiv1.Pod) bool {
 func hasNotSafeToEvictAnnotation(pod *apiv1.Pod) bool {
 	return pod.GetAnnotations()[PodSafeToEvictKey] == "false"
 }
+
+// PodShouldBeTerminated check if the pod should have already been remove by the system.
+// Pod has to be terminated, it must have run over its terminationGracePeriod, and also we add an extra time buffer of 'PodDeletionExtraBufferDuration'
+func PodShouldBeTerminated(pod *apiv1.Pod) bool {
+	if pod.DeletionTimestamp == nil {
+		return false
+	}
+
+	gracePeriod := pod.Spec.TerminationGracePeriodSeconds
+	if gracePeriod == nil {
+		defaultGracePeriod := int64(apiv1.DefaultTerminationGracePeriodSeconds)
+		gracePeriod = &defaultGracePeriod
+	}
+
+	if pod.DeletionTimestamp.Time.Add(time.Duration(*gracePeriod) * time.Second).Add(PodDeletionExtraBufferDuration).Before(time.Now()) {
```

Comment:
I'm not a fan of using `time.Now()` directly here, it makes the tests non-deterministic. I'd prefer to inject that as a parameter to the function, WDYT?

Reply:
yes, passing it as parameter is better. Doing it.